### PR TITLE
Empty player list on phase transition, despawn skulls, always reset weather

### PR DESCRIPTION
**Pull request by valaphee**
When the Java Client transitions between config to play, it resets the player list, and Velocity seems to not send player list remove packets. Resolves https://github.com/GeyserMC/Geyser/issues/4785

While I'm already there, skulls now despawn, for parity with normal entities, which also get despawned in Geyser when switching worlds/servers.

Improve https://github.com/GeyserMC/Geyser/issues/3679 as now at least the weather gets reset between dimension switches.

### CODE REVIEW

File: core/src/main/java/org/geysermc/geyser/session/GeyserSession.java [hunk 1]
```diff
@@ -887,8 +888,18 @@ private void connectDownstream() {
             public void packetReceived(Session session, Packet packet) {
                 if (protocol.getState() == ProtocolState.CONFIGURATION) {
                     if (packet instanceof ClientboundFinishConfigurationPacket) {
-                        // Prevent
-                        GeyserSession.this.ensureInEventLoop(() -> GeyserSession.this.sendDownstreamPacket(new ServerboundFinishConfigurationPacket()));
+                        GeyserSession.this.ensureInEventLoop(() -> {
```
Comment (Camotoy) [inline core/src/main/java/org/geysermc/geyser/session/GeyserSession.java:891]: Now that this is a bit more of an intensive method, the cleaner solution is probably to make a dedicated translator for this packet, and just "return" here so the finish configuration packet is sent on the session thread and not by MCProtocolLib.

Comment (valaphee) [reply core/src/main/java/org/geysermc/geyser/session/GeyserSession.java]: Reverted it, and added a separate translator

**Review by Camotoy — APPROVED** [review]

File: core/src/main/java/org/geysermc/geyser/translator/protocol/java/JavaFinishConfigurationPacketTranslator.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public void translate(GeyserSession session, ClientboundFinishConfigurationPacket packet) {
+        // Clear the player list, as on Java the player list is cleared after transitioning from config to play phase
+        PlayerListPacket playerListPacket = new PlayerListPacket();
+        playerListPacket.setAction(PlayerListPacket.Action.REMOVE);
+        for (PlayerEntity otherEntity : session.getEntityCache().getAllPlayerEntities()) {
+            playerListPacket.getEntries().add(new PlayerListPacket.Entry(otherEntity.getTabListUuid()));
+            session.getEntityCache().removePlayerEntity(otherEntity.getUuid());
+        }
```
Comment (Konicai) [inline core/src/main/java/org/geysermc/geyser/translator/protocol/java/JavaFinishConfigurationPacketTranslator.java:46]: `getAllPlayerEntities()` doesn't return a copy, and i'm not under the impression fastutil supports concurrent modification. i would add a `removeAllPlayerEntities()` method to `EntityCache`

File: core/src/main/java/org/geysermc/geyser/session/GeyserSession.java [hunk 3]
```diff
@@ -87,6 +87,7 @@
 import org.geysermc.geyser.entity.type.Entity;
 import org.geysermc.geyser.entity.type.ItemFrameEntity;
 import org.geysermc.geyser.entity.type.Tickable;
+import org.geysermc.geyser.entity.type.player.PlayerEntity;
```
Comment (Konicai) [inline core/src/main/java/org/geysermc/geyser/session/GeyserSession.java:90]: unused import

**Review by Konicai — CHANGES_REQUESTED** [review]

**Review by onebeastchris — APPROVED** [review]
lgtm!

**Comment by onebeastchris** [comment]
This will likely also resolve https://github.com/GeyserMC/Geyser/issues/4785

File: core/src/main/java/org/geysermc/geyser/session/cache/EntityCache.java [hunk 4]
```diff
@@ -141,6 +142,16 @@ public Collection<PlayerEntity> getAllPlayerEntities() {
...
 
+    public void removeAllPlayerEntities() {
+        PlayerListPacket playerListPacket = new PlayerListPacket();
+        playerListPacket.setAction(PlayerListPacket.Action.REMOVE);
+        for (PlayerEntity otherEntity : playerEntities.values()) {
+            playerListPacket.getEntries().add(new PlayerListPacket.Entry(otherEntity.getTabListUuid()));
+        }
+        session.sendUpstreamPacket(playerListPacket);
+        playerEntities.clear();
+    }
```
Comment (Konicai) [inline core/src/main/java/org/geysermc/geyser/session/cache/EntityCache.java:153]: I wouldn't send the packet here, nothing else in the class does

Comment (valaphee) [reply core/src/main/java/org/geysermc/geyser/session/cache/EntityCache.java:153]: spawnEntity, removeEntity, removeAllEntities removeBossBar and removeAllBossBars does indirectly

Comment (Konicai) [reply core/src/main/java/org/geysermc/geyser/session/cache/EntityCache.java:153]: yeah you're right, sorry for glossing over the rest

i'm mostly just concerned with keeping the PlayerEntity stuff cohesive (having all or none of the add/remove methods deal with packet sending)

**Review by Camotoy — APPROVED** [review]

**Review by Konicai — APPROVED** [review]

**Pull request merged**